### ISSUE TITLE
[Feature request] Make a fasd vim plugin

### ISSUE THREAD

**Issue by FreddieOliveira** [shell-fishes/meta#1]
It's been a while I'm searching for a fasd vim plugin and I've never found anything worth it. I think it would be nice to develop one. Maybe as a new repo. What you guys think?

**Comment by josephholsten** [comment]
Ooh, that sounds like a lovely complement to the FZF plugin I use.

**Comment by FreddieOliveira** [comment]
Indeed, the [FZF](https://github.com/junegunn/fzf.vim) plugin rocks. But, what I realized from my daily workflow is that I prefer to open my files from a terminal using fasd, instead of opening them from within vim. This causes me a problem, because I end up with a bunch of different vim sessions opened where I can't copy and paste between them.

More specifically I type `nvim ,fileName` and hit <kbd>Tab</kbd> so that the zle widget `fasd-complete` is executed.

I think it would be nice to also implement the plugin in a way it integrates with the FZF plugin, so the fasd possibilities are piped to it. The command can be named as simply :fasd, and if FZF plug is installed, it uses it:

![2020_09_30_102512](https://user-images.githubusercontent.com/9597536/94755782-2c146580-0385-11eb-8f7f-1ff79e7ae55e.png)

 Populating it with fasd results. Or, if it's not installed, we can use the built-in selection box (just like when you type `:e ` and hit <kbd>Tab</kbd>):

![Screenshot_20200930-222103_Termux](https://user-images.githubusercontent.com/9597536/94755889-87465800-0385-11eb-804e-9c5276387800.png)

Or open a new split with the results (like when you type `:e folder` and hit <kbd>Enter</kbd>):

![2020_09_30_102612](https://user-images.githubusercontent.com/9597536/94755944-b0ff7f00-0385-11eb-8751-d8553b134232.png)

**Comment by infokiller** [comment]
I think a vim plugin would be great. Note that vim has its own mechanism for storing information about previously opened files: viminfo (classic vim) and shada files (neovim). Both are loaded into `v:oldfiles`. I use the following functions in my dotfiles to read these files and select them using fzf:

https://github.com/infokiller/config-public/blob/c83926b9db0051894ab7f84e88e374e90bdc470a/.config/bash/functions.sh#L256-L321

I think a better approach than what I'm currently doing would be to have a vim plugin that updates the fasd database when a file is opened (and possibly load the `v:oldfiles` into the fasd database for the first time).

**Comment by FreddieOliveira** [comment]
fasd offers a manual way of adding entries with `fasd -A`. We could add some hook when opening a file and execute that command, so the database will be updated with files opened from within vim. We could do something similar with the `v:oldfiles`.
